extractor/text.go

```diff
@@ -721,18 +721,17 @@ func (to *textObject) renderText(data []byte) error {
 		spaceMetrics, _ = model.DefaultFont().GetRuneMetrics(' ')
 	}
 	spaceWidth := spaceMetrics.Wx * glyphTextRatio
-	common.Log.Trace("spaceWidth=%.2f text=%q font=%s fontSize=%.1f", spaceWidth, runes, font, tfs)
+	common.Log.Trace("spaceWidth=%.2f text=%q font=%s fontSize=%.2f", spaceWidth, runeSlices, font, tfs)
 
 	stateMatrix := transform.NewMatrix(
 		tfs*th, 0,
 		0, tfs,
 		0, state.trise)
 
-	common.Log.Trace("renderText: %d codes=%+v runes=%q", len(charcodes), charcodes, runes)
+	common.Log.Trace("renderText: %d codes=%+v runes=%q", len(charcodes), charcodes, len(runeSlices))
 
-	for i, r := range runes {
-		// TODO(peterwilliams97): Need to find and fix cases where this happens.
-		if r == '\x00' {
+	for i, r := range runeSlices {
+		if len(r) == 1 && r[0] == '\x00' {
```

Comment:
Did we identify in which case this happens?  a 0 input?  Typically in C means end of string or something.  Maybe a problem in the PDF that we can safely ignore.

Reply:
After not having to deal with this a long time I after I wrote the TODO, I looked in several open source PDF parsers and they all ignored this case.